src/Database/LSMTree/Internal.hs

```diff
@@ -424,29 +423,20 @@ openSession tr hfs hbio dir = do
         traceWith tr TraceNewSession
         FS.createDirectory hfs activeDirPath
         FS.createDirectory hfs snapshotsDirPath
-        mkSession sessionFileLock 0
+        mkSession sessionFileLock
 
     restoreSession sessionFileLock = do
         traceWith tr TraceRestoreSession
         -- If the layouts are wrong, we throw an exception, and the lock file
         -- is automatically released by bracketOnError.
         checkTopLevelDirLayout
+
+        FS.removeDirectoryRecursive hfs activeDirPath -- TODO: exceptions safety
+        FS.createDirectory hfs activeDirPath
```

Comment:
I should address this in this PR. Might be as simple as 

```suggestion
        FS.removeDirectoryRecursive hfs activeDirPath 
          `finally` FS.createDirectoryIfMissing False hfs activeDirPath
```

Reply:
Yes, otherwise if we fail in the middle, where no active dir exists, we can currently never open that session again.

Other possible approaches would be to only remove the content of the active dir or make `checkTopLevelDirLayout` more lenient (creating the the directory if it doesn't exist).

Reply:
Both alternatives are valid, though removing the dir contents is a bit more involved.  `listDirectory` just provides us with directory entry names and not the entry types (file/directory), which is something we'll have to figure out on the fly ourselves. I've gone with the `finally` approach for now.